### DISCUSSION TITLE
trying to deploy my first contract on zksync sepolia tesnet network

### DISCUSSION THREAD

**Discussion by QUEST09** [Cyfrin/foundry-full-course-cu#2632] [Q&A] [answered]
I keep get this error when I tried to deploy and verify the contract from the Cyfrin course. It keeps saying "solidity verification failed, logs can be read in the terminal log"
![image](https://github.com/user-attachments/assets/7ea5d397-e6b9-4bc7-bb18-af90a5e10630)
 here is my code 
 // SPDX-License-Identifier: MIT
pragma solidity ^0.8.24;

contract SimpleStorage {
    uint256 public myFavoriteNumber;
    
    struct Person {
        uint256 favoriteNumber;
        string name;
    }
    
    Person[] public listOfPeople;
    
    mapping(string => uint256) public nameToFavoriteNumber;

    function store(uint256 _favoriteNumber) public {
        myFavoriteNumber = _favoriteNumber;
    }

    function retrieve() public view returns(uint256) {
        return myFavoriteNumber;
    }

    function addPerson(string memory _name, uint256 _favoriteNumber) public {
        listOfPeople.push(Person(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber;
    }
}

**Comment by EngrPips** [comment, marked as answer]
Hello @QUEST09, Please create a `contracts` folder in the `default-workspace` and put your `SimpleStorage` file inside of that `contracts` folder then retry.

**Reply by QUEST09** [reply]
okay 
i did create that earlier on and the zksync plugin was able to compile it sir.

**Reply by EngrPips** [reply]
If you have done everything right and the issue persist, I will say leave it and proceed, as working with `zksync` isn't really smooth as of now and that is why every `zksync` content in the course is optional.

**Reply by QUEST09** [reply]
okay, thank you so much sir...
can I get your X handle so I can connect with you sir?

**Reply by EngrPips** [reply]
Yeah, sure, I am @EngrPips on Twitter

**Reply by oremnode** [reply]
Thank you so much, it worth it @EngrPips

**Reply by EngrPips** [reply]
Anytime, Friend. Keep crushing it.

**Comment by tamiromara** [comment]
Hi @QUEST09,
I encountered the same issue, and here's what worked for me:

I remove all the files that were generated by remix and recreated the contracts file and pasted the code back in the SimpleStorage.sol. 
Recompile, then deploy and don't forget to tick the "Verify Contract" box. 

Also, i renamed the file from "Contracts" to "contract" with small "c". Maybe it helped or maybe it didn't. But that's just what i did!

Good luck

**Comment by EngrPips** [comment]
Awesome. Thanks for the great heads-up.
